service/src/main/java/com/fecklessweasel/service/objectmodel/StoredFile.java

```diff
@@ -87,12 +90,13 @@ public static StoredFile create(Connection sql,
                                     String title,
                                     String description,
                                     String tag,
+                                    String extension,
```

Comment:
Must null check tag and extension with OM.nullCheck().

Also, you make the max length for the extension 4 in the database but you don't check it. Uploading a file called "foo.hello" would either silently truncate the extension, breaking the file storage, or it would throw an unhandled exception with an indecipherable message.